pkg/imgpkg/bundle/images_refs.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package bundle
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	ctlimg "github.com/k14s/imgpkg/pkg/imgpkg/image"
+	"github.com/k14s/imgpkg/pkg/imgpkg/lockconfig"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/cppforlife/go-cli-ui/ui.UI
+
+type ImageRef struct {
+	lockconfig.ImageRef
+	IsBundle *bool
+}
+
+func NewImageRef(imgRef lockconfig.ImageRef, isBundle bool) ImageRef {
+	return ImageRef{ImageRef: imgRef, IsBundle: &isBundle}
+}
+
+func (i ImageRef) DeepCopy() ImageRef {
+	var isBundle *bool
+	if i.IsBundle != nil {
+		tmp := *i.IsBundle
+		isBundle = &tmp
+	}
+	return ImageRef{
+		ImageRef: i.ImageRef.DeepCopy(),
+		IsBundle: isBundle,
+	}
+}
+
+func (i *ImageRef) updateRepo(relativeToRepo string) *ImageRef {
+	i.Image = replaceImageRepo(i.Image, relativeToRepo)
+	return i
+}
+
+type ImageRefs struct {
+	refs                 []ImageRef
+	imageLocationsConfig *ImageLocationsConfig
+	originalImagesLock   *lockconfig.ImagesLock
+
+	refsLock *sync.Mutex
+}
+
+func NewImageRefs() ImageRefs {
+	return ImageRefs{
+		refsLock: &sync.Mutex{},
+	}
+}
+
+func NewImageRefsFromImagesLock(imagesLock lockconfig.ImagesLock, imageRefsLocationConfig ImageRefLocationsConfig) (ImageRefs, error) {
+	imageRefs := ImageRefs{
+		refsLock:           &sync.Mutex{},
+		originalImagesLock: &imagesLock,
+	}
+	for _, lockImgRef := range imagesLock.Images {
+		imageRefs.AddImagesRef(ImageRef{
+			ImageRef: lockImgRef,
+			IsBundle: nil,
+		})
+	}
+
+	err := imageRefs.syncImageRefsWithLocationConfig(imageRefsLocationConfig)
+	if err != nil {
+		return ImageRefs{}, err
+	}
+	return imageRefs, nil
+}
+
+func (i *ImageRefs) LocalizeToRepo(relativeToRepo string) {
+	i.refsLock.Lock()
+	defer i.refsLock.Unlock()
+
+	for j, imgRef := range i.refs {
+		i.refs[j].AddLocation(replaceImageRepo(imgRef.Image, relativeToRepo))
+	}
+}
+
+func (i *ImageRefs) UpdateRelativeToRepo(imgRetriever ctlimg.ImagesMetadata, relativeToRepo string) (bool, error) {
+	if i.imageLocationsConfig != nil {
+		i.LocalizeToRepo(relativeToRepo)
+		return true, nil
+	}
+
+	for _, ref := range i.refs {
+		image, err := name.ParseReference(ref.updateRepo(relativeToRepo).Image)
```

Comment:
so should this change to just using replaceImageRepo and then we remove updateRepo() method?

Reply:
ok done.